src/server/contextListeners.ts

```diff
@@ -0,0 +1,72 @@
+/**
```

Comment:
nit: let's call this `instrumentation.ts`. Also, unfortunately, this needs rebase.